src/foam/nanos/boot/NSpecFactory.java

```diff
@@ -46,10 +47,18 @@ public Object create(X x) {
       try {
         if (logger != null) logger.info("Creating Service", spec_.getName());
         ns_ = spec_.createService(x_.getX().put(NSpec.class, spec_));
-        if (logger != null) logger.info("Created Service", spec_.getName());
-
-        if (ns_ instanceof ContextAware) ((ContextAware) ns_).setX(x_.getX());
-        if (ns_ instanceof NanoService) ((NanoService) ns_).start();
+        Object ns = ns_;
+        while ( ns != null ) {
+          if (ns instanceof ContextAware) ((ContextAware) ns).setX(x_.getX());
+          if (ns instanceof NSpecAware) ((NSpecAware) ns).setNSpec(spec_);
+          if (ns instanceof NanoService) ((NanoService) ns).start();
+          if (ns instanceof ProxyDAO) {
```

Comment:
What is this for?

Reply:
Previously, this test and set only occured on the top spec.  But in serviceScript, if you wrapped your dao in proxyDAO or a DAO which was not NSpecAware, then the 'set' never occurred. EasyDAO is NSpecAware and this makes sure it gets set.  This also came up during simple clustering setup where the NSpec was needed but not set because only the top dao was inspected.

Reply:
Should we then call dao.cmd(spec_), then any DAO's in the chain that are interested can get the NSpec?